From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct ahci_host_priv
Patch-mainline: never, kabi

In 3.12.51, commit 4f348a8d38f7bf2a4137acbeed2057485b6fd618 (libahci:
Allow drivers to override start_engine), upstream commit
039ece38da45f5e6a94be3aa7611cf3634bc2461 added one member to struct
ahci_host_priv. It indeed changed the layout of that structure.

Since it is a private structure to ahci and allocated internally, we
can just protect the change the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/ata/ahci.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/ata/ahci.h
+++ b/drivers/ata/ahci.h
@@ -328,7 +328,9 @@ struct ahci_host_priv {
 	 * default ahci_start_engine during ahci_save_initial_config, this can
 	 * be overridden anytime before the host is activated.
 	 */
+#ifndef __GENKSYMS__
 	void			(*start_engine)(struct ata_port *ap);
+#endif
 };
 
 extern int ahci_ignore_sss;
